File: public/dsa/searching/matrix_search/naive/matrix_search.py
from typing import List


class Solution:
    def searchMatrix(self, matrix: List[List[int]], key: int) -> bool:
        rowSize = len(matrix)
        columnSize = len(matrix[0])
        for i in range(rowSize):
            for j in range(columnSize):
                if matrix[i][j] == key:
                    return True
        return False


# tests-start
matrix1 = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9],
]

matrix2 = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9],
    [10, 11, 12],
]

sol = Solution()
print(sol.searchMatrix(matrix1, 6))
print(sol.searchMatrix(matrix2, 15))

# tests-end


"""output
True
False
"""

File: public/dsa/searching/matrix_search/naive/matrix_search.cpp
#include <bits/stdc++.h>
using namespace std;

bool searchMatrix(vector<vector<int>> &matrix, int key) {
    int rowSize = matrix.size(), columnSize = matrix[0].size();
    for (int i = 0; i < rowSize; i++) {
        for (int j = 0; j < columnSize; j++) {
            if (matrix[i][j] == key) {
                return true;
            }
        }
    }
    return false;
}

// tests-start
int main() {
    vector<vector<int>> matrix1 = {
        {1, 2, 3},
        {4, 5, 6},
        {7, 8, 9},
    };

    vector<vector<int>> matrix2 = {
        {1, 2, 3},
        {4, 5, 6},
        {7, 8, 9},
        {10, 11, 12},
    };

    cout << boolalpha << searchMatrix(matrix1, 6) << endl;
    cout << boolalpha << searchMatrix(matrix2, 15) << endl;

    return 0;
}

// tests-end

/*output
true
false
*/